### ISSUE TITLE
Create-React-App fails with: "Failed to minify the code from this file: ./node_modules/jsts/extend.js:2"

### ISSUE THREAD

**Issue by jnachtigall** [bjornharrtell/jsts#357]
I am using jsts as part of Create-React-App together with OpenLayers. Works so far following the hints at #323 (that is, using static functions like with `import BufferOp from 'jsts/org/locationtech/jts/operation/buffer/BufferOp';`.

This works in development mode, but when I run the production build step it fails:
```console
$ yarn run build
yarn run v1.6.0
$ react-scripts build
Creating an optimized production build...
Failed to compile.

Failed to minify the code from this file:

        ./node_modules/jsts/extend.js:2

Read more here: http://bit.ly/2tRViJ9

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "build-js" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

1. The given link http://bit.ly/2tRViJ9 hints at the point that jsts is wrongly published to npmjs. There are some issues regarding ES6 modules / webpack that say that something is broken, e.g. #311. I am not sure how to go forward here.
2. Well, actually, it does not even work in IE11 when in development mode. Error is `syntax error`:
```js
/***/ "./node_modules/jsts/java/lang/Character.js":
/*!**************************************************!*\
  !*** ./node_modules/jsts/java/lang/Character.js ***!
  \**************************************************/
/*! exports provided: default */
/*! exports used: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
/* harmony export (immutable) */ __webpack_exports__["a"] = Character;
function Character () {}
Character.isWhitespace = c => ((c <= 32 && c >= 0) || c == 127)
```

This is a arrow function. Not sure why it is like this although webpack together with babel is used.

Problems only exist when using jsts package. I have a dozen of other dependencies where it works.

**Comment by bjornharrtell** [comment]
Yep, still not sure how to fix this. Contributions with suggested fixes has as of yet been of the nature "this fix may or may not work" and with potential regressions so I'm still looking for a proper solution.

**Comment by jnachtigall** [comment]
Sorry, I can not spent time on this, also my knowledge on webpack and stuff is limited. I think the whole es6 stuff is a bit broken here, for instance, the point that I need to use a static Buffer function; or that the build sizes are hugh:
![image](https://user-images.githubusercontent.com/3307404/45034669-6d2bd580-b058-11e8-9af5-7ff884aab9fc.png)

TBH, I was just looking for a buffer function for some distance based filtering and in the end it's probably easier to migrate to http://turfjs.org/docs#buffer :-/

**Comment by bjornharrtell** [comment]
It is huge because of most operations having a very complex internal dependency hierarchy which is partially due to the original code being quite ancient Java.

For buffer, Turf actually uses JSTS because no suitable simpler implementation has yet been found/written. Well, actually they now use an unofficial fork of JSTS with manually hacked source that perhaps can reduce the bundle size in the case of webpack. See https://github.com/DenisCarriere/turf-jsts. Essentially the hack uses ES6 classes which I also wanted to use but I could not make it work with some of the functionality of JSTS - perhaps functionality ignored or not used by Turf. The issues of duplicated modules in bundle is however likely a webpack drawback because I've not seen the same issue when using Rollup.

**Comment by bjornharrtell** [comment]
This actually got me interested in trying to go ES6 classes in the transpilation again. :)

**Comment by bjornharrtell** [comment]
Aside from that I've just made a 1.6.2 release without an empty index.js module. Not sure if it will help or if it will cause other issues and would appreciate any feedback from testing it with webpack and elsewhere.

**Comment by jnachtigall** [comment]
Thanks, very much appreciated. 

I upgraded to 1.6.2. However, it still fails with the very same error message 
```
Failed to minify the code from this file:

        ./node_modules/jsts/extend.js:2
```

CRE is pretty common and easy to setup. So I think it would a good sandbox for your testing. It also has a pretty common default webpack and babel configuration, so jsts it works there it probably works everywhere. Just follow the 3 steps in https://github.com/facebook/create-react-app#quick-overview:
```
npx create-react-app my-app
cd my-app
```
Then add some jsts code and import to App.js and run `npm run build` (or `yarn run build`) and you should see the error.

**Comment by jnachtigall** [comment]
fwiw, I migrated to turf/buffer now. For the reasons you mentioned, built size is still huge, but it works. Both building and development mode under IE11.

![image](https://user-images.githubusercontent.com/3307404/45086243-00b9e080-b103-11e8-8203-98dcc8fc29ce.png)

(also, ergonomics of turf are a bit better: no static function import, no OL3Parser and - what looked strange to me - injecting of ol objects)

**Comment by bjornharrtell** [comment]
Thanks, I'll keep this issue open because it would be good if I can get the basic CRE setup working at some point.

Agreed turf has done well in the ergonomics department and for some operations it means it brings JSTS functionality to a wider audience. JSTS is really just a one-man hack that I wanted for my own interests but it became more widely known and used/depended on than I could imagine.. :)

**Comment by bjornharrtell** [comment]
Had some transpilation fun, source is now fully ES6 classes and released as 2.0.0. Limited external testing as of yet (certainly not the CRE setup) but all 524 local tests pass.

**Comment by bjornharrtell** [comment]
Assuming this is resolved. Reopen if proven otherwise.
